lib/statsd/instrument/backends/store_backend.rb

```diff
@@ -0,0 +1,73 @@
+require 'monitor'
+require 'time'
+require 'fileutils'
+
+module StatsD::Instrument::Backends
+  class StoreBackend < StatsD::Instrument::Backend
+    MAXIMUM_NAME_LENGTH = 256
+    TYPE_PACKS = {
+      c:  'Qq',
+      ms: 'Qf',
+      g:  'Qf',
+      h:  'Qf',
+    }
+    MetricNameStruct = Struct.new(:id, :name, :type) do
+      def to_bin
+        raise "Metric name cannot be longer than #{MAXIMUM_NAME_LENGTH}" if name.size > MAXIMUM_NAME_LENGTH
+        [id, name, type.to_s].pack("QA#{MAXIMUM_NAME_LENGTH}A2")
+      end
+    end
+    MetricStruct = Struct.new(:id, :value, :type) do
```

Comment:
You definitely want to include `sample_rate` here (at least when it is not equal to 1). This allows you to know that we suppressed a bunch of events when analyzing the data.

Personally, I'd also like it if tags would the included as well.